Indicators/IndicatorResult.cs

```diff
@@ -0,0 +1,53 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// Represents the result of an indicator's calculations
+    /// </summary>
+    public class IndicatorResult
+    {
+        /// <summary>
+        /// The indicator output value
+        /// </summary>
+        public decimal Value
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The indicator status
+        /// </summary>
+        public IndicatorStatus Status
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IndicatorResult"/> class
+        /// </summary>
+        /// <param name="value">The value output by the indicator</param>
+        /// <param name="status">The status returned by the indicator</param>
+        public IndicatorResult(decimal value, IndicatorStatus status = IndicatorStatus.Success)
```

Comment:
@mchandschuh No problem, I'll remove the optional, I thought that future usage inside indicators looked less verbose:

`return new IndicatorResult(value)`
vs
`return new IndicatorResult(value, IndicatorStatus.Success)`